engine/schema/src/main/resources/META-INF/db/schema-41200to41300.sql

```diff
@@ -19,3 +19,29 @@
 -- Schema upgrade from 4.12.0.0 to 4.13.0.0
 --;
 
+CREATE TABLE `cloud`.`template_ovf_properties` (
+  `id` bigint(20) unsigned NOT NULL AUTO_INCREMENT,
+  `template_id` bigint(20) unsigned NOT NULL,
+  `key` VARCHAR(100) NOT NULL,
+  `type` VARCHAR(45) DEFAULT NULL,
+  `value` VARCHAR(100) DEFAULT NULL,
+  `qualifiers` TEXT DEFAULT NULL,
+  `user_configurable` TINYINT(1) NOT NULL DEFAULT '0',
+  `label` TEXT DEFAULT NULL,
+  `description` TEXT DEFAULT NULL,
+  `removed` datetime DEFAULT NULL,
+  PRIMARY KEY (`id`),
+  CONSTRAINT `fk_template_ovf_properties__template_id` FOREIGN KEY (`template_id`) REFERENCES `vm_template`(`id`)
+) ENGINE=InnoDB DEFAULT CHARSET=utf8;
+
+CREATE TABLE `cloud`.`user_vm_ovf_properties` (
```

Comment:
@nvazquez can't we not reuse the _details table for this? Essentially the properties can be treated as properties of the template and the vm instance?

Reply:
Fair enough, but we'll need to distinguish them from other VM details. Prepending a predefined string to each key?

Reply:
@nvazquez that that is fine, you can define an enum or constant string somewhere to use it across both template and vm details table. The value can be a json or list that you can read/load as needed.

Reply:
Done, thanks